packages/devtools_app/lib/src/flutter/flex_split_column.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'package:flutter/material.dart';
+
+import 'split.dart';
+
+class FlexSplitColumn extends StatelessWidget {
+  FlexSplitColumn({
+    Key key,
+    @required this.totalHeight,
+    @required this.headers,
+    @required List<Widget> children,
+    @required List<double> initialFractions,
+    List<double> minSizes,
+  })  : assert(children != null && children.length >= 2),
+        assert(initialFractions != null && initialFractions.length >= 2),
+        assert(children.length == initialFractions.length),
+        adjustedChildren = adjustChildren(children, headers),
+        adjustedInitialFractions =
+            adjustInitialFractions(initialFractions, headers, totalHeight),
+        adjustedMinSizes = adjustMinSizes(minSizes, headers),
+        super(key: key) {
+    if (minSizes != null) {
+      assert(minSizes.length == children.length);
+    }
+  }
+
+  /// The headers that will be laid out above each corresponding child in
+  /// [children].
+  ///
+  /// All headers but the first will be passed into [Split.splitters] when
+  /// creating the [Split] widget in `build`. Instead of being passed into
+  /// [Split.splitters], it will be combined with the first child in [children]
+  /// to create the first child we will pass into [Split.children].
+  ///
+  /// We do this because the first header will not actually be a splitter as
+  /// there is not any content above it for it to split.
+  ///
+  /// We adjust other values [adjustedChildren], [adjustedInitialFractions], and
+  /// [adjustedMinSizes] to account for the first header. We do this adjustment
+  /// here so that the creators of [FlexSplitColumn] can be unaware of
+  /// the under-the-hood calculations necessary to achieve the UI requirements
+  /// specified by [initialFractions] and [minSizes].
+  final List<FlexSplitColumnHeader> headers;
+
+  /// The children that will be laid out below each corresponding header in
+  /// [headers].
+  ///
+  /// All [children] except the first will be passed into [Split.children]
+  /// unmodified. We need to adjust the first child from [children] to account
+  /// for the first header (see above).
+  final List<Widget> adjustedChildren;
+
+  /// The fraction of the layout to allocate to each child in [children].
+  ///
+  /// We need to adjust the values given by [initialFractions] to account for
+  /// the first header (see above).
+  final List<double> adjustedInitialFractions;
+
+  /// The minimum size each child is allowed to be.
+  final List<double> adjustedMinSizes;
+
+  /// The total height of the column, including all [headers] and [children].
+  final double totalHeight;
+
+  @visibleForTesting
+  static List<Widget> adjustChildren(
```

Comment:
the name adjust is confusing as this method doesn't have side effects. maybe a method name that indicates what the widget list it is building looks like. E.g. buildSomethingSomething

Reply:
changed to `buildChildrenWithFirstHeader`